### PR TITLE
Upgrade CableTray

### LINKED ISSUE #161: Add support for CableTray modifications

### PR DESCRIPTION
**Pull request by michaelhoehn**
### NOTE: Depends on 

https://github.com/BHoM/BHoM/pull/1248

   
### Issues addressed by this PR

Closes #161

### CODE REVIEW

**Review by tiagogrossi — APPROVED** [review]
Solution built and seems to be working as intended in a test script.

**Comment by michaelhoehn** [comment]
@BHoMBot check core

**Comment by michaelhoehn** [comment]
@BHoMBot check copyright compliance

**Comment by michaelhoehn** [comment]
@BHoMBot check copyright-compliance

**Comment by michaelhoehn** [comment]
@BHoMBot check core

**Comment by michaelhoehn** [comment]
@BHoMBot check installer

**Comment by michaelhoehn** [comment]
@BHoMBot check compliance

**Comment by michaelhoehn** [comment]
@BHoMBot check versioning

File: BHoMUpgrader43/Converter.cs [hunk 1]
```diff
@@ -38,18 +38,41 @@ public class Converter : Base.Converter
...
 
-        
+            Dictionary<string, object> newVersion = new Dictionary<string, object>(oldVersion);
 
-        /***************************************************/
+            if (newVersion.ContainsKey("StartPoint"))
+                newVersion.Add("_t", "BH.oM.Geometry.Point");
+
+            if (newVersion.ContainsKey("EndPoint"))
+                newVersion.Add("_t", "BH.oM.Geometry.Point");
```
Comment (FraserGreenroyd) [inline BHoMUpgrader43/Converter.cs:61]: I don't think this will do what you're expecting?

File: BHoMUpgrader43/Converter.cs [hunk 2]
```diff
@@ -38,18 +38,41 @@ public class Converter : Base.Converter
...
+
+            if (newVersion.ContainsKey("SectionProperty"))
+                newVersion.Add("SectionProperty", null);
 
+            if (newVersion.ContainsKey("OrientationAngle"))
+                newVersion.Add("OrientationAngle", 0);
+
+            if (newVersion.ContainsKey("ConnectionProperty"))
+                newVersion.Remove("ConnectionProperty");
```
Comment (FraserGreenroyd) [inline BHoMUpgrader43/Converter.cs:70]: I think this is the only bit we actually need?

File: BHoMUpgrader43/Converter.cs [hunk 3]
```diff
@@ -38,18 +38,41 @@ public class Converter : Base.Converter
...
 
-        /***************************************************/
+            if (newVersion.ContainsKey("StartPoint"))
+                newVersion.Add("_t", "BH.oM.Geometry.Point");
+
+            if (newVersion.ContainsKey("EndPoint"))
+                newVersion.Add("_t", "BH.oM.Geometry.Point");
+
+            if (newVersion.ContainsKey("SectionProperty"))
+                newVersion.Add("SectionProperty", null);
```
Comment (FraserGreenroyd) [inline BHoMUpgrader43/Converter.cs:64]: This will overwrite any existing SectionProperty to be null

File: BHoMUpgrader43/Converter.cs [hunk 4]
```diff
@@ -38,18 +38,41 @@ public class Converter : Base.Converter
...
 
+            if (newVersion.ContainsKey("OrientationAngle"))
+                newVersion.Add("OrientationAngle", 0);
```
Comment (FraserGreenroyd) [inline BHoMUpgrader43/Converter.cs:67]: Likewise this will overwrite any existing OrientationAngle to be 0

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
```
private static Dictionary<string, object> UpgradeCableTray(Dictionary<string, object> oldVersion)
        {
            if (oldVersion == null)
                return null;

            Dictionary<string, object> newVersion = new Dictionary<string, object>(oldVersion);

            if (newVersion.ContainsKey("ConnectionProperty"))
                newVersion.Remove("ConnectionProperty");

            return newVersion;
        }
```

I think this is all the method needs to contain given the changes on the object?

**Comment by michaelhoehn** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright-compliance

**Review by FraserGreenroyd — APPROVED** [review]
Reviewed through with @michaelhoehn and all is good now.

**Pull request merged**